lib/util/itinerary.js

```diff
@@ -390,21 +390,55 @@ export function getLegIcon (leg, customIcons) {
   }
   let iconStr = leg.mode
   if (iconStr === 'CAR' && leg.rentedCar) {
-    iconStr = leg.rentedCarData.companies[0]
+    iconStr = leg.from.networks[0]
   } else if (iconStr === 'CAR' && leg.tncData) {
     iconStr = leg.tncData.company
   } else if (iconStr === 'BICYCLE' && leg.rentedBike) {
-    // placeholder for future vehicle rental refactor
-  } else if (iconStr === 'MICROMOBILITY' && leg.rentedVehicle && leg.rentedVehicleData) {
-    iconStr = leg.rentedVehicleData.companies[0]
+    iconStr = leg.from.networks[0]
+  } else if (iconStr === 'MICROMOBILITY' && leg.rentedVehicle) {
+    iconStr = leg.from.networks[0]
   }
 
   return getIcon(iconStr, customIcons)
 }
 
-export function getPlaceName (place) {
+export function getCompanyForNetwork (networkString, companies = []) {
+  const company = companies.find(co => co.id === networkString)
+  if (!company) {
+    console.warn(`No company found in config.yml that matches rented vehicle network: ${networkString}`, companies)
+  }
+  return company
+}
+
+export function getModeStringForCompany (company) {
```

Comment:
Since a company can offer multiple modes, this may not be future-proof. Can't we just use the mode of the leg to get the label?

Reply:
@evansiroky, after trying this out, I don't think this proposed approach really works that well. Basically, as I mentioned on MS Teams, we would need to check the mode on the next leg in order to determine what vehicle/place the user is walking towards, which would require a larger refactor and feels a little scattered. Perhaps, instead of using the company though, we could check against the vertex type, which will tell us whether it is a car, bike, or "vehicle". How does this sound?